font-codegen/src/table.rs

```diff
@@ -349,14 +349,17 @@ pub(crate) fn generate_group_compile(
     let docs = &item.attrs.docs;
     let name = &item.name;
     let inner = &item.inner_type;
-    //let type_field = &item.inner_field;
 
     let mut variant_decls = Vec::new();
     let mut write_match_arms = Vec::new();
     let mut validate_match_arms = Vec::new();
     let mut from_obj_match_arms = Vec::new();
     let from_type = quote!(#parse_module :: #name);
+    let mut first_var_name = None;
     for var in &item.variants {
+        if first_var_name.is_none() {
+            first_var_name = Some(&var.name);
+        }
```

Comment:
I'm failing to grasp why this needs to be an Option and set in the loop. I imagine a GenericGroup with no variants is meaningless so what would be wrong with just writing something like `let first_var_name = &item.variants.first().unwrap().name;`?

Reply:
yes fair point.